engine/src/main/java/com/odysseusinc/arachne/executionengine/auth/BigQueryCredentialsProvider.java

```diff
@@ -0,0 +1,64 @@
+package com.odysseusinc.arachne.executionengine.auth;
+
+import com.google.common.collect.ImmutableMap;
+import com.odysseusinc.arachne.commons.types.DBMSType;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.DataSourceUnsecuredDTO;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+@Service
+@Slf4j
+public class BigQueryCredentialsProvider implements CredentialsProvider {
+	private static final String RUNTIME_BQ_KEYFILE = "BQ_KEYFILE";
+
+	@Override
+	public AuthEffects apply(DataSourceUnsecuredDTO dataSource, Path analysisDir, String analysisDirInContainer) {
+		if (dataSource.getType() == DBMSType.BIGQUERY) {
```

Comment:
The formatting looks really weird here. Did you use tabulation?

Reply:
Fixed